sdk/servicebus/service-bus/src/connectionContext.ts

```diff
@@ -346,11 +490,34 @@ export namespace ConnectionContext {
       if (context.connection.isOpen()) {
         log.ns("Closing the amqp connection '%s' on the client.", context.connectionId);
 
-        // Close all the clients.
-        for (const id of Object.keys(context.clientContexts)) {
-          const clientContext = context.clientContexts[id];
-          await clientContext.close();
+        // Close all the senders.
+        for (const senderName of Object.keys(context.senders)) {
+          await context.senders[senderName].close();
+        }
+
+        // Close batching receiver
+        for (const batchingReceiverName of Object.keys(context.batchingReceivers)) {
+          await context.batchingReceivers[batchingReceiverName].close();
```

Comment:
Is it still useful to keep batching and streaming receivers in separate lists?

Reply:
The`onDetached()` method on the streaming and batching receivers have different signatures. If we consolidate them, then yes, we don't have to track them as separate lists. We can do that as a follow up after this PR is merged.

Reply:
Logged #10619